### ISSUE TITLE
Intrinsic function names are not validated in Conditions or Outputs

### ISSUE THREAD

**Issue by trav-c** [aws-cloudformation/cfn-python-lint#1221]
*cfn-lint version:*  0.25.3

*Description of issue.*
It appears that cfn lint does not check the name of intrinsic functions used in either the Conditions or the Outputs section of the template.

Using "Fn::Of": / !Of instead of "Fn::Or": / !Or in either the Conditions or Outputs section of the template is not detected as an error, while trying to use it in the Resources section correctly triggers an error (E3002  Property ... has an illegal function Fn::Of

The following template generates E3002 for Resources/SSM/Properties/Value but does not detect / report any problems in either the Conditions or Outputs sections, both of which reference the non-existent intrinsic functions 'Of' and Offer

```
Description: "Test - unknown intrinsic"
Parameters:
    TestParam1:
        Type: String
        Default: Value1
    TestParam2:
        Type: String
        Default: Value2


Conditions:

    HasParam:
      "Fn::Of":
        - !Not [ !Equals [ !Ref TestParam1, '' ] ]
        - !Not [ !Equals [ !Ref TestParam2, '' ] ]

Resources:

    SSM:
        Type: AWS::SSM::Parameter
        Condition: HasParam
        Properties:
            Type: String
            Description: "Just a test"
            Name: "/This-Is-Just-A-Test"
            Value: !Of TestParam1

Outputs:

    Suffix:
        Description: This is a test
        Value: !Offer TestParam1
```

**Comment by kddejong** [comment]
Agreed.  Let me work on something for this.  Thanks for reporting it.

**Comment by kddejong** [comment]
Half of this is done.  Need to do this for outputs still.

**Comment by kddejong** [comment]
Should be all there in the next release.
